numpy/core/tests/test_indexing.py

```diff
@@ -404,6 +404,22 @@ def __getitem__(self, item):
         arr = np.zeros((1,), dtype=[('f1', 'i8'), ('f2', 'i8')])
         assert_array_equal(arr[SequenceLike()], arr[SequenceLike(),])
 
+    def test_indexing_array_weird_strides(self):
+        # See also gh-6221
+
+        C = 4
+        x = np.ones(10)
+        x2 = np.ones((10, 2))
+        ind = np.arange(10)[:, None, None, None]
+        ind = np.broadcast_to(ind, (10, 55, C, C))
+
+        # single advanced index case
+        assert_array_equal(x[ind], x[ind.copy()])
+        # higher dimensional advanced index
+        zind = np.zeros(4, dtype=np.intp)
```

Comment:
Wouldn't it make sense to use `C` instead of `4` here, like you do in the definition of `ind`?